pkg/pilot/elasticsearch/v5/controller.go

```diff
@@ -1,11 +1,41 @@
 package v5
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
 )
 
+const (
+	elasticsearchLocalNodeID = "_local"
+)
+
 func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
-	if p.genericPilot.IsThisPilot(pilot) {
+	// TODO: perform cluster wide actions if we are a leader
+	if !p.genericPilot.IsThisPilot(pilot) {
+		return nil
+	}
+	if pilot.Status.Elasticsearch == nil {
+		pilot.Status.Elasticsearch = &v1alpha1.ElasticsearchPilotStatus{}
+	}
+	// set the ES document count to nil to indicate an unknown number of
+	// documents in case obtaining the document count fails. This prevents the
+	// node being shut down if it is not safe to do so.
+	pilot.Status.Elasticsearch.Documents = nil
```

Comment:
So the supplied `Pilot` gets modified, even if this function errs?

Reply:
Yes